platform/javascript/serve.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+from http.server import HTTPServer, SimpleHTTPRequestHandler, test
+import os
+import sys
+import logging
+import argparse
+import subprocess
+
+
+class CORSRequestHandler(SimpleHTTPRequestHandler):
+    def end_headers(self):
+        self.send_header("Cross-Origin-Opener-Policy", "same-origin")
+        self.send_header("Cross-Origin-Embedder-Policy", "require-corp")
+        SimpleHTTPRequestHandler.end_headers(self)
+
+
+def shell_open(url):
+    if sys.platform == "win32":
+        os.startfile(url)
+    else:
+        opener = "open" if sys.platform == "darwin" else "xdg-open"
+        subprocess.call([opener, url])
+
+
+if __name__ == "__main__":
+    # Change to the directory where the script is located,
+    # so that the script can be run from any location.
+    os.chdir(os.path.dirname(os.path.realpath(__file__)))
```

Comment:
how about keeping this close to `if args.root: os.chdir(args.root)` lower, since it's related together. Maybe it can be there in `else:`?

Reply:
```suggestion
    os.chdir(Path(__file__).resolve().parent)
```

together with `from pathlib import Path`